### ISSUE TITLE
the `local` problem

### ISSUE THREAD

**Issue by glycerine** [gijit/gi#1]
from issue 9:

Variables in Lua are global by default. The `local` prefix makes a variable local to the current scope.
~~~
# in luajit:
> local a = 1 -- local variable, only alive for this line
> =a
nil
> a = 3 -- global variable
> =a
3
> 
~~~
However, at the REPL, entering one line, the repl treats that line as the entire scope.

Hence local variables automatically disappear after their line executes.

So if you are single stepping through a function, the `gi` Go-to-Lua translator would need to generate different code if it is generating a whole function at once (where `local` is useful for scoping), versus generating code for interactive step through (where `local` makes your variable disappear); or `gi` would need to forego `local` variables altogether for some other scoping mechanism.

Probably have to do the 2nd choice. We could keep a current local scope around, and by default read and write to that.
 ~~~
> curScope={}
> curScope.a = 1
> =curScope.a
1
> curScope.a = 2
> =curScope.a
2
> 
~~~
 
here's how scope stack management would look
~~~
> cur={}
> cur.a=1
> cur={par=cur} -- push a new scope onto the top of the stack
> cur.b=2
> cur.a=3
> =cur.par.a
1
> =cur.a
3
> cur = cur.par -- pop the top of the scope stack
> =cur.a
1
>
~~~

**Comment by glycerine** [comment]
> gi translator would need to generate different code if it is generating a whole function at once (where local is useful for scoping)

This first alternative actually looks easier and simpler now. It's not that difficult to tell if your are at package scope or inside a function. 

I'm working on this now.

**Comment by glycerine** [comment]
fixed in v0.7.0
